cypress/support/cypress-commands/lifecycle.js

```diff
@@ -293,7 +293,19 @@ Cypress.Commands.add('invokeLifecycleApi', (appId, method, methodParams = null)
     params,
     additionalParams
   );
-  cy.log(CONSTANTS.LIFECYCLE_INTENT + JSON.stringify(publishMessage));
+
+  let publishMessageForLog;
+  if (publishMessage && publishMessage.data && publishMessage.data.query) {
+    publishMessageForLog = publishMessage.data.query;
+    if (typeof publishMessageForLog == 'string') {
+      publishMessageForLog = JSON.parse(publishMessageForLog);
+    }
+    if (publishMessageForLog.params.methodName == 'Lifecycle.history') {
+      cy.log(CONSTANTS.LIFECYCLE_HISTORY_INTENT + JSON.stringify(publishMessage));
+    } else {
+      cy.log(CONSTANTS.LIFECYCLE_INTENT + JSON.stringify(publishMessage));
+    }
+  }
```

Comment:
cant we use this list 
CONSTANTS.LIFECYCLE_APIS to determine if its a lifecycle api or history ?

Reply:
updated to     if (publishMessageForLog.params.methodName == CONSTANTS.LIFECYCLE_APIS.HISTORY) {